crates/ibc/src/core/ics03_connection/handler/conn_open_try.rs

```diff
@@ -4,117 +4,123 @@ use crate::core::ics03_connection::connection::{ConnectionEnd, Counterparty, Sta
-    let expected_conn = ConnectionEnd::new(
-        State::Init,
-        msg.counterparty.client_id().clone(),
-        Counterparty::new(msg.client_id.clone(), None, ctx.commitment_prefix()),
+    let version = ctx.pick_version(
+        ctx.get_compatible_versions(),
         msg.counterparty_versions.clone(),
+    )?;
+
+    let self_connection_end = ConnectionEnd::new(
+        State::TryOpen,
+        msg.client_id.clone(),
+        msg.counterparty.clone(),
+        vec![version],
         msg.delay_period,
     );
 
-    // 2. Pass the details to the verification function.
-    verify_proofs(
-        ctx,
-        msg.client_state,
-        msg.proofs.height(),
-        &new_connection_end,
-        &expected_conn,
-        &msg.proofs,
-    )?;
+    {
+        let client_state = ctx.client_state(self_connection_end.client_id())?;
+        let consensus_state =
+            ctx.client_consensus_state(self_connection_end.client_id(), msg.proofs_height)?;
 
-    // Transition the connection end to the new state & pick a version.
-    new_connection_end.set_state(State::TryOpen);
+        {
+            let counterparty_connection_id = self_connection_end
+                .counterparty()
+                .connection_id()
+                .ok_or_else(Error::invalid_counterparty)?;
+            let counterparty_expected_connection_end = ConnectionEnd::new(
+                State::Init,
+                msg.counterparty.client_id().clone(),
+                Counterparty::new(msg.client_id.clone(), None, ctx.commitment_prefix()),
+                msg.counterparty_versions.clone(),
+                msg.delay_period,
+            );
 
-    // Pick the version.
-    new_connection_end.set_version(ctx.pick_version(
-        ctx.get_compatible_versions(),
-        msg.counterparty_versions.clone(),
-    )?);
+            client_state
+                .verify_connection_state(
+                    msg.proofs_height,
+                    self_connection_end.counterparty().prefix(),
+                    &msg.proof_connection_end,
+                    consensus_state.root(),
+                    counterparty_connection_id,
+                    &counterparty_expected_connection_end,
+                )
+                .map_err(Error::verify_connection_state)?;
+        }
 
-    assert_eq!(new_connection_end.versions().len(), 1);
+        client_state
+            .verify_client_full_state(
+                msg.proofs_height,
+                self_connection_end.counterparty().prefix(),
+                &msg.proof_client_state,
+                consensus_state.root(),
+                self_connection_end.counterparty().client_id(),
+                msg.client_state,
+            )
+            .map_err(|e| {
+                Error::client_state_verification_failure(self_connection_end.client_id().clone(), e)
+            })?;
 
-    output.log("success: connection verification passed");
```

Comment:
It seems like we removed all logs, would be nice to keep them.

Reply:
I removed them because they were inconsistent with ibc-go (e.g. [connOpenAck](https://github.com/cosmos/ibc-go/blob/c8e2ebcf2eaa6b487ca52499063c7c3db4253398/modules/core/03-connection/keeper/handshake.go#L232)). We should open an issue for that.

Until we fix that, do we prefer "custom logs" or no logs at all?

Reply:
I think logs are allowed to be non-deterministic, so it wouldn't hurt to keep them for now.

Reply:
[6931db8](https://github.com/cosmos/ibc-rs/pull/170/commits/6931db89b5082886c8392e9b8c53377bc03ef74c)